### DISCUSSION TITLE
GUI for pyaeroval config files

### DISCUSSION THREAD

**Discussion by AugustinMortier** [metno/pyaerocom#684] [Ideas]
I just drop the idea here:
Making a GUI for creating the config files for pyaeroval.

This would allow the users to:
- know which data are available (meaning we need to store this information somewhere: database?)
- know all the options available
- be aware of default values
- prevent from creating a broken config file

For info, I once started a [draft page](https://aeroval.met.no/lab.php) for making a config file online, just so you know what I'm talking about by GUI.

**Comment by avaldebe** [comment]
Hi @AugustinMortier 

I like this idea, but I'm concerned that til will end up as a half finished unsupported side project.

> This would allow the users to:
> 
> * know which data are available (meaning we need to store this information somewhere: database?)
> * know all the options available
> * be aware of default values
> * prevent from creating a broken config file

Implementing pydantic models for the pyaeroval configuration, would address the last 3 points.
It could be used as a the back end of a configuration GUI or configuration checker CLI.
Also, modern IDEs (like VScode and PyCharm) can use the pydantic models to provide completions.
